server/lib/tuist/utilities/date_formatter.ex

```diff
@@ -108,4 +108,53 @@ defmodule Tuist.Utilities.DateFormatter do
   end
 
   def format_iso(_), do: "Unknown"
+
+  @doc """
+  Format datetime with timezone conversion.
+
+  Takes a UTC datetime and converts it to the specified timezone.
+  Falls back to UTC display if timezone is nil or conversion fails.
+
+  ## Examples
+
+      iex> DateFormatter.format_with_timezone(~U[2024-01-15 14:30:25Z], "America/New_York")
+      "Mon 15 Jan 09:30:25 EST"
+
+      iex> DateFormatter.format_with_timezone(~U[2024-01-15 14:30:25Z], nil)
+      "Mon 15 Jan 14:30:25 UTC"
+  """
+  def format_with_timezone(%DateTime{} = datetime, timezone) when is_binary(timezone) do
+    local_time = Timex.Timezone.convert(datetime, timezone)
+    Timex.format!(local_time, "{WDshort} {D} {Mshort} {h24}:{m}:{s}")
+  rescue
+    _ ->
+      # Fallback to UTC if timezone conversion fails
```

Comment:
When would this happen? Should we report it to AppSignal?

Reply:
> When would this happen? Should we report it to AppSignal?

The timezone is sourced from JavaScript, so there be dragons. I'd rather handle this scenario when we get an unexpected timezone.